main.go

```diff
@@ -148,14 +154,18 @@ func registerEHR(server *echo.Echo, config Config, customerRepository customers.
 		log.Fatal(err)
 	}
 
-	fhirClientFactory := fhir.NewFactory(fhir.WithURL(config.FHIR.Server.Address))
+	fhirClientFactory := fhir.NewFactory(fhir.WithURL(config.FHIR.Server.Address), fhir.WithMultiTenancyEnabled(config.FHIR.Server.SupportsMultiTenancy()))
 	patientRepository := patients.NewFHIRPatientRepository(patients.Factory{}, fhirClientFactory)
 	transferRepository := transfer.NewSQLiteTransferRepository(sqlDB)
 	orgRegistry := registry.NewOrganizationRegistry(&nodeClient)
 	vcRegistry := registry.NewVerifiableCredentialRegistry(&nodeClient)
 	transferService := transfer.NewTransferService(authService, fhirClientFactory, transferRepository, customerRepository, orgRegistry, vcRegistry)
 	tenantInitializer := func(tenant string) error {
-		return fhir.InitializeTenant(config.FHIR.Server.Type, config.FHIR.Server.Address, tenant)
+		if !config.FHIR.Server.SupportsMultiTenancy() {
```

Comment:
intead of: if !canFoo { return }; foo() -> if canFoo { foo() } ? Allows for other logic after the call to InitializeTenant.